nemo/collections/nlp/data/information_retrieval/gpt_embedding_dataset.py

```diff
@@ -244,9 +256,12 @@ def collate_fn(self, batch):
                 lengths.append(len(item['query']))
                 input_ids.append(item['pos_doc'])
                 lengths.append(len(item['pos_doc']))
-                input_ids.append(item['neg_doc'])
-                lengths.append(len(item['neg_doc']))
-                max_length = max(max_length, len(item['query']), len(item['pos_doc']), len(item['neg_doc']))
+                for nd in item['neg_doc']:
+                    input_ids.append(nd)
+                    lengths.append(len(nd))
+                max_length = max(
+                    max_length, len(item['query']), len(item['pos_doc']), *(len(nd) for nd in item['neg_doc'])
+                )
```

Comment:
can we push the max_lenght book-keeping for neg_docs into the loop above? and set it once before the loop for query and pos_doc? it would be easier to read.
before the loop:
`max(max_length, len(item['query']), len(item['pos_doc']))`
In the loop:
`max(max_length, len(nd))`

Reply:
Done